core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3564,10 +3564,20 @@ private SyncResult syncInodeMetadata(RpcContext rpcContext, LockedInodePath inod
         if (ufsFpParsed.isValid()) {
           short mode = Short.parseShort(ufsFpParsed.getTag(Tag.MODE));
           long opTimeMs = System.currentTimeMillis();
-          setAttributeSingleFile(rpcContext, inodePath, false, opTimeMs, SetAttributeContext
-              .mergeFrom(SetAttributePOptions.newBuilder().setOwner(ufsFpParsed.getTag(Tag.OWNER))
-                  .setGroup(ufsFpParsed.getTag(Tag.GROUP)).setMode(new Mode(mode).toProto()))
-              .setUfsFingerprint(ufsFingerprint));
+          SetAttributePOptions.Builder builder = SetAttributePOptions.newBuilder()
+              .setMode(new Mode(mode).toProto());
+          String owner = ufsFpParsed.getTag(Tag.OWNER);
+          if (!owner.equals(Fingerprint.UNDERSCORE)) {
```

Comment:
I am curious, under what conditions, would the ufs not have owner or group information?

Reply:
its's a corner case, took me a while to figure out. i'll update the description

Reply:
@yuzhu when inherit_acl is false all object ufs do not return owner and group. that is because the s3 account credential cannot be mapped to a user in those cases.

Reply:
got it.